optax/_src/linear_algebra.py

```diff
@@ -81,12 +83,12 @@ def _iter_body(state):
 
 
 def matrix_inverse_pth_root(
-    matrix,
-    p,
-    num_iters=100,
-    ridge_epsilon=1e-6,
-    error_tolerance=1e-6,
-    precision=lax.Precision.HIGHEST):
+    matrix: chex.Array,
+    p: int,
+    num_iters: Optional[int]=100,
+    ridge_epsilon: Optional[float]=1e-6,
+    error_tolerance: Optional[float]=1e-6,
+    precision: Optional[jnp.float32]=lax.Precision.HIGHEST):
```

Comment:
Afaik `lax.Precision` is an Enum. That means we could very nicely enforce the 3 available options (DEFAULT, HIGH, HIGHEST) by typing this as `lax.Precision`.